### PR TITLE
[Maps] Adding Operations for airQuality (current, forecast/daily/ and forecast/hourly) to weather swagger

### PR DESCRIPTION
**Pull request by john35452**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/john35452/azure-rest-api-specs/dev-maps-Microsoft.Maps-1.0...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ambientlight** [comment]
@john35452: This needs to be structured accordingly to https://github.com/Azure/azure-rest-api-specs/pull/14871 since Weather service will be under its own client.

**Comment by john35452** [comment]
> @john35452: This needs to be structured accordingly to #14871 since Weather service will be under its own client.

Thanks for the remind! Currently, this PR is used to check schema with PM for now.
We will move files to weather client directory after #14871 is merged.

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 1]
```diff
@@ -702,6 +709,203 @@
...
+          {
+            "name": "query",
+            "description": "Coordinates of the location for which air quality is requested. The applicable query is specified as a comma separated string composed by latitude followed by longitude e.g. \"47.641268,-122.125679\".",
+            "in": "query",
+            "type": "string",
+            "required": true
+          },
+          {
+            "name": "duration",
+            "description": "Specifies for how many days from now we would like to know about the air quality. Available values are 1 ~ 4.",
```
Comment (evaou) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:808]: Per below AccuWeather API doc, the available time spans are 1, 2, 3, 4, 5, 6, and 7 days.
http://apidev.accuweather.com/developers/airQualityAPIParameters#queryParameters

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 2]
```diff
@@ -702,6 +709,203 @@
...
+          },
+          "500": {
+            "$ref": "#/responses/500"
+          }
+        }
+      }
+    },
+    "/weather/airQuality/forecast/daily/{format}": {
+      "get": {
+        "description": "**Get Daily Air Quality Forecast**\n\n**Applies to**: S1 pricing tiers.\n\nThe service provides the detailed information about the concentration of pollutant and overall status for upcoming few days air quality. The service can provide forecast air quality for future one to four days.",
```
Comment (evaou) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:776]: To follow existing naming convention, we could update the API title from "Get Daily Air Quality Forecast" to "Get Air Quality Daily Forecast".

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 3]
```diff
@@ -702,6 +709,203 @@
...
+          },
+          "500": {
+            "$ref": "#/responses/500"
+          }
+        }
+      }
+    },
+    "/weather/airQuality/forecast/hourly/{format}": {
+      "get": {
+        "description": "**Get Hourly Air Quality forecast**\n\n**Applies to**: S1 pricing tiers.\n\nThe service provides the detailed information about the concentration of pollutant and overall status for current air quality.",
```
Comment (evaou) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:844]: To follow existing naming convention, we could update the API title from "Get Hourly Air Quality Forecast" to "Get Air Quality Hourly Forecast".

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json (hunk 1 above)
Comment (john35452) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:808]: Thanks for the remind!
It's kind of surprising that our current implementation only allows 1 to 4 which we also need to take care of in the near future.
https://msazure.visualstudio.com/One/_git/Azure-IoT-LocationBasedServices?path=%2Fsrc%2FALSProxy%2FALS%2FModels%2FWeather%2FDailyForecastAirQualityRequest.cs&_a=contents&version=GBmaster

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json (hunk 2 above)
Comment (john35452) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json]: Updated

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json (hunk 3 above)
Comment (john35452) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json]: Updated

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 4]
```diff
@@ -76,6 +76,13 @@
     }
   },
   "parameters": {
+    "Pollutants": {
+      "name": "Pollutants",
+      "in": "query",
+      "type": "string",
```
Comment (ambientlight) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:82]: the description mentions this is boolean?

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 5]
```diff
@@ -702,6 +709,203 @@
...
+            "name": "query",
+            "description": "Coordinates of the location for which air quality is requested. The applicable query is specified as a comma separated string composed by latitude followed by longitude e.g. \"47.641268,-122.125679\".",
+            "in": "query",
+            "type": "string",
+            "required": true
+          },
+          {
+            "name": "duration",
+            "description": "Specifies for how many days from now we would like to know about the air quality. Available values are 1, 2, 3, 4, 5, 6, and 7.",
+            "default": 1,
```
Comment (sataneja) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:809]: We should mention that the default value is 1 in the description itself. Marking "default" property = 1 may help the SDK's but it doesn't show up in the final documentation so it's not easily visible to the user.

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 6]
```diff
@@ -76,6 +76,13 @@
     }
   },
   "parameters": {
+    "Pollutants": {
+      "name": "Pollutants",
```
Comment (sataneja) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:80]: 'p'ollutants

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json (hunk 4 above)
Comment (sataneja) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:82]: it'll work either way as our web server will translate the string value to boolean when we hydrate the request query object but swagger does support a boolean type so we can use that as well. See if we have any other API that has a boolean and if we use type = string or boolean. As i said it'll work either way at the service level so atleast we can be consistent

Comment (ambientlight) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:82]: booleans are way better for generated clients

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 7]
```diff
@@ -702,6 +709,203 @@
...
+          },
+          {
+            "name": "duration",
+            "description": "Specifies for how many days from now we would like to know about the air quality. Available values are 1, 2, 3, 4, 5, 6, and 7.",
+            "default": 1,
+            "in": "query",
+            "type": "integer"
+          },
+          {
+            "$ref": "#/parameters/Pollutants"
```
Comment (sataneja) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:814]: Just double check but i looked at the Accuweather docs and I don't think the "pollutants" query param is supported for the "daily" API. It's only supported for "hourly" and "current" API's - http://apidev.accuweather.com/developers/airQualityAPIGuide#dailyFcastv2 - It makes sense if you see the response but i wd just run the daily api in Postman to confirm and if not just remove it from the Daily API docs

Comment (john35452) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:814]: Thanks for the remind! After I try with the postman. This parameter is not supported in daily forecasat, so I remove it now.

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json (hunk 6 above)
Comment (john35452) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:80]: Updated

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json (hunk 5 above)
Comment (john35452) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:809]: Thanks for the remind. I have added it on.

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json [hunk 8]
```diff
@@ -76,6 +76,13 @@
     }
   },
   "parameters": {
+    "Pollutants": {
+      "name": "pollutants",
+      "in": "query",
+      "type": "boolean",
+      "required": false,
+      "description": "Boolean value that returns detailed information about each pollutant. By default is True."
```
Comment (ambientlight) [inline specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json:84]: @john35452: `"x-ms-parameter-location": "method"` is also needed, this is a method parameter right?

**Comment by ambientlight** [comment]
> Thanks for the remind! Currently, this PR is used to check schema with PM for now.
We will move files to weather client directory after #14871 is merged.

@john35452: Also keep in mind the changes introduced:
1. Common definitions were moved into https://github.com/Azure/azure-rest-api-specs/blob/master/specification/maps/data-plane/Common/preview/1.0/common.json, so ClientId, ResponseFormat(JSONFormat) parameters in path should reference the common definitions.
2. SubscriptionKey is not needed as explicit parameter and is rather set in securityDiffinitions: https://github.com/Azure/azure-rest-api-specs/blob/0b8ad3639180cfd04c880544275f18a6601c2226/specification/maps/data-plane/Weather/preview/1.0/weather.json#L32-L37

File: specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json (hunk 4 above)
Comment (ambientlight) [reply specification/maps/data-plane/Microsoft.Maps/Weather/preview/1.0/weather.json]: @sataneja: I think we should update others to bool too in another PR for consistency.

I mean, generated code is like:

```go
alertsResp, err := weatherClient.GetSevereWeatherAlerts(ctx, weather.ResponseFormatJSON, "48.057,-81.091", &weather.WeatherGetSevereWeatherAlertsOptions{
   Details:  to.StringPtr("true"),
   Language: to.StringPtr("EN"),
})
```
Having string parameters for bools is weird from clients perspective. Generated bools are better:

```go
currentAirQualityResp, err := weatherClient.GetCurrentAirQuality(ctx, weather.ResponseFormatJSON, "47.632346,-122.138874", &weather.WeatherGetCurrentAirQualityOptions{
    Language:   to.StringPtr("EN"),
    Pollutants: to.BoolPtr(true),
})
```

**Comment by ambientlight** [comment]
@john35452: thanks, saw the https://github.com/Azure/azure-rest-api-specs/pull/15351

**Comment by anuchandy** [comment]
@JeffreyRichter - the service is already deployed and this pr adds three 3 endpoints to swagger to align with the service. 

In an offline sync map engineer mentioned - the APIs for Azure Maps have been live for a few years, hence these changes aren't really breaking - the service APIs haven't changed, and there are no SDK released.

**Comment by anuchandy** [comment]
@@JeffreyRichter could you provide the breaking change approval if there are no concerns?

**Comment by JeffreyRichter** [comment]
This does seem like a bunch of breaking changes. Is this matching service behavior or does the serviced want to make this change in which case it should be done in a new api-version?

**Comment by JeffreyRichter** [comment]
I approved this because I just read your first note. But, this seems like a bunch of breaking changes pushed on to customers. Service teams shouldn't just deploy new API contract changes and then update their swagger saying that this is NOW matching service behavior in order to get approval.

**Comment by ambientlight** [comment]
@anuchandy, @JeffreyRichter: this should not be merged, superseded by https://github.com/Azure/azure-rest-api-specs/pull/15351

**Comment by JeffreyRichter** [comment]
So, should I approve #15351?
If so, I'm hesitant to because of the removed property. See https://github.com/Azure/azure-rest-api-specs/pull/15351/checks?check_run_id=3218460533

**Comment by ambientlight** [comment]
@JeffreyRichter:

To clarify, the removal of host parametrization is intentional (https://github.com/Azure/azure-rest-api-specs/pull/15351/files#diff-434f5529f9a095a3d6a4e2520ab5b7067f710bcb75ce41788533eb6fcf0976e2R7), as it was added by mistake previously and should not appear in map clients except creator. 

However this has been already addressed in a seperate PR: https://github.com/Azure/azure-rest-api-specs/pull/15386, we can remove it here in order not cause [1007 - RemovedClientParameter](https://github.com/Azure/openapi-diff/blob/master/docs/rules/1007.md) and then there should not be any other concerns to approve https://github.com/Azure/azure-rest-api-specs/pull/15351 ?

**Comment by john35452** [comment]
Since this PR is targeted at Microsoft.Map directory which is in the old directory structure, and we need to add our new change to client directory now.
So I raise another PR for this change.
https://github.com/Azure/azure-rest-api-specs/pull/15351
https://portal.azure-devex-tools.com/branch/john35452/azure-rest-api-specs/dev-maps-Weather-1.0...master/
Thanks a lot and sorry for confusing!

**Pull request closed without merging**